### DISCUSSION TITLE
conference 2021

### DISCUSSION THREAD

**Discussion by DenisYaroshevskiy** [jfalcou/eve#795] [Done & Done!]
Pitching plan: https://github.com/jfalcou/eve/wiki/conference-2021

**Comment by DenisYaroshevskiy** [comment]
What's left?

* Elevator pitch: we show comparisons between std::find_if, std::remove_if and eve::find_if, eve::remove_if on my machine.
   DONE

* Why are we better? SIMD. Crash course into SIMD architectures: from sse2 - to avx512, arm neon. Some measurements between architectures.
 Needed inclusive scan ported: https://github.com/jfalcou/eve/issues/894
 I dunno how `compress_store` will do on sse2 - probably extremly bad.
 Which is maybe fare or we should try things.
 
 * What's eve? 3 layers: core, math, algo.
    No code needed.
    
* What basic algos we have at the time? (I feel we need at least all algos I had last year: find_if, transform_reduce/reduce, inclusive_scan).
   find_if
   mismatch
   reduce
   hack transform reduce/skip transform reduce/ do it proper?
   inclusive scan algorithm?
 
* How they are implemented under the hood? Show case strlen
   Done
 
 * How to collect indexes
   Mostly done
 
 * ZIP: running algorithms on parallel arrays
Just show  that mismatch is implemented with find_if I think. 

* Objects: our object structure, soa_vector, convert between objects on cartesian coordinate planes to polar (or maybe something better - just a pretty real example).

* Algorithm support for product types.
References for products?
soa_vector basic: push_back, erase (needs eve::algo::copy?), scalar iterator
At this point remove won't with product types. Needs: https://github.com/jfalcou/eve/issues/868

* More about customisation points: remove spaces using a manually implemented crazy x86 trick.
Not done but simple

* (Stretch goal): Build stuff. How one can build a library in C++20 with eve for different architectures and integrate it with the C++11 code.
Not gonna happen

**Reply by jfalcou** [reply]
Are we going to shoot everything at once to the 1st conference or can we spread that over C++Con, MC++, C++Russia  and CPPPP ?

**Reply by DenisYaroshevskiy** [reply]
I think one talk given multiple times.
objects will blow the mind and we need to get there.
